vmware-tanzu/cartographer
# Issue 57/reject templates with namespace

Resolves #57

Adds validating webhook for
* `clusterconfigtemplates`
* `clusterimagetemplates`
* `clustersourcetemplates`
* `clustertemplates`

To test this behavior:
1. add `metadata.namespace` to the templates in the example `suppy-chain-templates.yaml`
2. apply the templates to the cluster
3. observe the following error messages
```
> k apply -f app-operator

clustersupplychain.carto.run/supply-chain created
Error from server (invalid template: template should not set metadata.namespace on the child object): error when creating "app-operator/supply-chain-templates.yaml": admission webhook "source-template-validator.cartographer.com" denied the request: invalid template: template should not set metadata.namespace on the child object
Error from server (invalid template: template should not set metadata.namespace on the child object): error when creating "app-operator/supply-chain-templates.yaml": admission webhook "image-template-validator.cartographer.com" denied the request: invalid template: template should not set metadata.namespace on the child object
Error from server (invalid template: template should not set metadata.namespace on the child object): error when creating "app-operator/supply-chain-templates.yaml": admission webhook "template-validator.cartographer.com" denied the request: invalid template: template should not set metadata.namespace on the child object
```

Questions:
1. Are we okay with the text of the error message?
2. I held off on adding other validations that seemed potentially sensible (e.g. does the stamped object have a kind?). Do we want more validations and if so, should we create another issue or add them to this PR?
